src/main/app/src/components/common/CookieModal.js

```diff
@@ -0,0 +1,251 @@
+
+  const { data, isLoading } = useContentful();
+
+  const contentfulTexts = single(data.cookieModalText);
+
+  const fields = {
+    shortContent: contentfulTexts['shortContent'] || '',
+    fullContent: contentfulTexts['fullContent'] || '',
+    heading: contentfulTexts['heading'] || t('cookieModal.heading'),
+    expandLinkText: contentfulTexts['expandLinkText'] || '',
+    settingsButtonText:
+      contentfulTexts['settingsButtonText'] || t('cookieModal.settings'),
+    settingsButtonCloseText:
+      contentfulTexts['settingsButtonCloseText'] || t('cookieModal.settings'),
+    acceptButtonText: contentfulTexts['acceptButtonText'] || t('cookieModal.accept'),
+    settingsHeaderText: contentfulTexts['settingsHeaderText'] || '',
+    settingsAcceptMandatoryText:
+      contentfulTexts['settingsAcceptMandatoryText'] || t('cookieModal.mandatory'),
+    settingsAcceptStatisticText:
+      contentfulTexts['settingsAcceptStatisticText'] || t('cookieModal.statistic'),
+    settingsAcceptMarketingText:
+      contentfulTexts['settingsAcceptMarketingText'] || t('cookieModal.marketing'),
+  };
+
+  function handleAcceptCookies(e) {
+    e.preventDefault();
+    Cookies.set(mandatoryCookieName, 'true', {
+      expires: 1800,
+      path: '/',
+    });
+    if (statisticCookiesAccepted) {
+      Cookies.set('oph-statistic-cookies-accepted', 'true', {
+        expires: 1800,
+        path: '/',
+      });
+    }
+    if (marketingCookiesAccepted) {
+      Cookies.set('oph-marketing-cookies-accepted', 'true', {
+        expires: 1800,
+        path: '/',
+      });
+    }
+    setCookiesAccepted(true);
+  }
+
+  const openSettings = (
+    <div id="cookie-modal-settings" className={classes.settings}>
+      <Divider className={classes.divider} />
+      <Typography variant="h3" className={classes.settingsHeader}>
+        {fields.settingsHeaderText}
+      </Typography>
+      <div>
+        <KonfoCheckbox
+          id="mandatoryCookies"
+          className={classes.settingsCheckbox}
+          checked
+          disabled
+        />
+        <label htmlFor="mandatoryCookies">{fields.settingsAcceptMandatoryText}</label>
+      </div>
+      <div>
+        <KonfoCheckbox
+          id="statisticCookies"
+          className={classes.settingsCheckbox}
+          checked={statisticCookiesAccepted}
+          disableripple
+          onClick={() => setStatisticCookiesAccepted(!statisticCookiesAccepted)}
+        />
+        <label htmlFor="statisticCookies">{fields.settingsAcceptStatisticText}</label>
+      </div>
+      <div>
+        <KonfoCheckbox
+          id="marketingCookies"
+          className={classes.settingsCheckbox}
+          checked={marketingCookiesAccepted}
+          disableripple
+          onClick={() => setMarketingCookiesAccepted(!marketingCookiesAccepted)}
+        />
+        <label htmlFor="marketingCookies">{fields.settingsAcceptMarketingText}</label>
```

Comment:
Noi labelit tosiaan vois vielä antaa childina tolle checkbox-komponentille
```suggestion
        <KonfoCheckbox
          id="mandatoryCookies"
          className={classes.settingsCheckbox}
          checked
          disabled
        >
          {fields.settingsAcceptMandatoryText}
        </KonfoCheckbox>
      </div>
      <div>
        <KonfoCheckbox
          id="statisticCookies"
          className={classes.settingsCheckbox}
          checked={statisticCookiesAccepted}
          disableripple
          onClick={() => setStatisticCookiesAccepted(!statisticCookiesAccepted)}
        >
          {fields.settingsAcceptStatisticText}
        </KonfoCheckbox>
      </div>
      <div>
        <KonfoCheckbox
          id="marketingCookies"
          className={classes.settingsCheckbox}
          checked={marketingCookiesAccepted}
          disableripple
          onClick={() => setMarketingCookiesAccepted(!marketingCookiesAccepted)}
        >
          {fields.settingsAcceptMarketingText}
        </KonfoCheckbox>
```

Reply:
Ainakaan suoraan noin se ei toimi, ei näy mitään labelia tällä tavalla. Onkohan jossain vielä joku vipu jota pitää vääntää jos haluaa tehdä noin?